pkg/apis/v1beta1/nodepool.go

```diff
@@ -202,3 +210,72 @@ func (pl *NodePoolList) OrderByWeight() {
 		return ptr.Int32Value(pl.Items[a].Spec.Weight) > ptr.Int32Value(pl.Items[b].Spec.Weight)
 	})
 }
+
+// GetAllowedDisruptions returns the minimum allowed disruptions across all disruption budgets for a given node pool.
+// This returns two values as the resolved value for a percent depends on the number of current node claims.
+func (in *NodePool) GetAllowedDisruptions(ctx context.Context, c clock.Clock) (intstr.IntOrString, intstr.IntOrString, error) {
+	vals := make([]intstr.IntOrString, len(in.Spec.Disruption.Budgets))
+	errs := make([]error, len(in.Spec.Disruption.Budgets))
+	workqueue.ParallelizeUntil(ctx, len(in.Spec.Disruption.Budgets), len(in.Spec.Disruption.Budgets), func(i int) {
+		val, err := in.Spec.Disruption.Budgets[i].GetAllowedDisruptions(c)
+		if err != nil {
+			errs[i] = fmt.Errorf("invalid budget %s, %w", lo.FromPtr(in.Spec.Disruption.Budgets[i].Crontab), err)
+		}
+		vals[i] = val
+	})
+	if err := multierr.Combine(errs...); err != nil {
+		return intstr.IntOrString{}, intstr.IntOrString{}, err
+	}
+	minIntVal, minPercentVal := math.MaxInt64, math.MaxInt64
+	for i := range vals {
+		val := vals[i]
+		// The crontab wasn't active
+		if val.IntVal == -1 {
+			continue
+		}
+		// This returns the percent value if it's a string, and the raw value if it's an int.
+		temp, err := intstr.GetScaledValueFromIntOrPercent(lo.ToPtr(val), 100, false)
+		if err != nil {
+			// Should almost never happen since this is validated when the nodepool is applied
+			return intstr.IntOrString{}, intstr.IntOrString{}, fmt.Errorf("getting intstr scaled value, %w", err)
+		}
+		if val.Type == intstr.Int {
+			minIntVal = lo.Ternary(temp < minIntVal, temp, minIntVal)
+		} else {
+			minPercentVal = lo.Ternary(temp < minPercentVal, temp, minPercentVal)
+		}
+	}
+	// return the values, defaulting to -1 if the value is MaxInt64
+	return intstr.FromInt(lo.Ternary(minIntVal == math.MaxInt64, -1, minIntVal)), intstr.FromString(fmt.Sprintf("%d%%", lo.Ternary(minPercentVal == math.MaxInt64, -1, minPercentVal))), nil
+}
+
+// GetAllowedDisruptions returns an intstr.IntOrString that can be used a comparison
+// for calculating if a disruption action is allowed. It returns an error if the
+// crontab is invalid. This returns -1 if the value is unbounded.
+func (b *Budget) GetAllowedDisruptions(c clock.Clock) (intstr.IntOrString, error) {
+	active, err := b.IsActive(c)
```

Comment:
When does this error? Is there ever a valid case where this errors out?

Reply:
It errors out when the the cron parser catches something that validation didn't. Since regex is notorious for having edge cases, I'd like to keep this here regardless of how fleshed out our validation is to ensure that we're always getting a schedule that's valid.